app/repos/ConsumerRepo.scala

```diff
@@ -3,40 +3,48 @@ package repos
 import com.google.inject.Inject
 import domain.Consumer
 import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
-import slick.jdbc.JdbcProfile
 import slick.jdbc.PostgresProfile.api._
-import slick.lifted.TableQuery
-import utils.VendorProxyConfig
+import slick.jdbc.{GetResult, JdbcProfile, SetParameter}
 
+import java.sql.Types
+import java.util.UUID
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class ConsumerRepo @Inject() (val dbConfigProvider: DatabaseConfigProvider, val config: VendorProxyConfig)
+class ConsumerRepo @Inject() (val dbConfigProvider: DatabaseConfigProvider)
     extends HasDatabaseConfigProvider[JdbcProfile] {
 
-  lazy val ConsumersTable = TableQuery[ConsumersTable]
-
-  def persist(c: Consumer): Future[Consumer] =
-    db.run(ConsumersTable returning ConsumersTable.map(_.id) into ((c, id) => c.copy(id)) += c)
-
-  def deleteByName(name: String): Future[Int] = db.run(ConsumersTable.filter(_.name === name).delete)
+  implicit def helpersSlickGetResultUUID: GetResult[UUID] =
+    GetResult(r => r.nextObject.asInstanceOf[UUID])
+
+  implicit def slickSetParameterUUID: SetParameter[UUID] =
+    SetParameter { case (v, pp) => pp.setObject(v, Types.OTHER) }
+
+  def persist(c: Consumer): Future[UUID] = {
+    val credentialId = UUID.randomUUID()
+    db.run(
+        for {
+          _ <- sqlu"INSERT INTO credentials(id, key, token, owner) VALUES ($credentialId, ${c.id}, ${c.token}, ${c.name})"
```

Comment:
Does this run in a transaction?

Reply:
It seems the for-comprehension would need to be wrapped with a call to `.transactionally`, something like

```
(for {
   ...
} yield().transactionally)
```